katcp/resource_client.py

```diff
@@ -391,8 +391,22 @@ def list_sensors(self, filter="", strategy=False, status="",
             dict.items(self.sensor), filter, strategy, status, use_python_identifiers, tuple, refresh)
 
     @tornado.gen.coroutine
-    @steal_docstring_from(resource.KATCPResource.set_sensor_strategy)
```

Comment:
I thought that is what you will say :-)
The problem was that the current implementation of set_sampling_strategy in the Container class had a bug and it set the strategy on each and every child resource. It just happened to work because the right one was amongst those. To fix it without too much ripple for now, I had to change the footprint for set_sampling_strategy in the Container class to include the resource_name. 
The sharing of the KATCPResource base class for KATCPClientResource and KATCPClientResourceContainer is somewhat awkward in some cases and causes unexpected side-effects or introduces un-tested use cases which are now started to get used (because it is available) in code which shows up bugs and anomalies between KATCPClientResource and KATCPClientResourceContainer.  
[Aargh - that just sounds unclear - better to have a chat about this some time.]